### PR TITLE
Remove the traefik addon dependency for jenkins

### PR DESCRIPTION
**Pull request by takirala**

### CODE REVIEW

File: metadata/static/jenkins/overview.md [hunk 1]
```diff
@@ -6,7 +6,7 @@ ways to define your build pipelines and integrating with a large number of testi
 
 ## Support Level
 - Mixed workload tested
-  - 100 Jenkins Masters
+  - 250 Jenkins Masters
```
Comment (takirala) [inline metadata/static/jenkins/overview.md]: We tested upto 500 jenkins masters but after 250 masters, the launch duration exceeded 600 seconds (which was used as a cutoff for DC/OS jenkins mwt)

Comment (shaneutt) [reply metadata/static/jenkins/overview.md]: Thanks for testing this :+1:

File: addons/jenkins/1.x/jenkins.yaml [hunk 2]
```diff
@@ -21,9 +21,6 @@ spec:
       enabled: true
     - name: none
       enabled: true
-  requires:
-    - matchLabels:
-        kubeaddons.mesosphere.io/provides: traefik
```
Comment (shaneutt) [inline addons/jenkins/1.x/jenkins.yaml]: In my own tests, I have not seen a NEED for traefik, though it's nice to have. I'm :heavy_check_mark: on removing this req.

File: metadata/static/jenkins/overview.md (hunk 1 above)
Comment (shaneutt) [reply metadata/static/jenkins/overview.md]: :heavy_check_mark:

**Comment by shaneutt** [comment]
Just a heads up, CI is broken here currently as we're waiting on fixes to KUDO (see [DCOS-62466](https://jira.mesosphere.com/browse/DCOS-62466)).

**Comment by shaneutt** [comment]
Just a heads up: I needed this for https://github.com/mesosphere/kubeaddons-enterprise/pull/13 so I cherry picked it in.

**Comment by shaneutt** [comment]
CI issues should now be resolved, let me know if you run into any further troubles with the CI testing code.

**Comment by takirala** [comment]
@shaneutt can you merge this PR ? (i don't have merge rights)

**Pull request merged**